client/js/controllers/editor.js

```diff
@@ -1036,6 +1038,130 @@
                     }
                 };
 
+                $scope.getNodes = function () {
+                    $http.get('/api/nodetypes/get')
+                    .success(function (data) {
+                        if (data) {
+                            for (var i in data) {
+                                for (var j in data[i]) {
+                                    nodenames.push(data[i][j].name);
+                                    inports.push(data[i][j].in_ports);
+                                    outports.push(data[i][j].out_ports);
+                                    if (data[i][j].options) {
+                                        members.push(data[i][j].options.members);
+                                        membernodes.push(data[i][j].name);
+                                    }
+                                }
+                            }
+                        }
+                    }).error(function (data) {
+                        console.log("Error getting nodetypes: " + data);
+                    });
+                };
+
+                $scope.getNodes();
+
+                var prettifyNodetype = function (nodetype) {
+                    var temp = nodetype.split('/');
+                    temp[0] = temp[0].toUpperCase();
+                    if (temp[1]) {
+                        temp[1] = temp[1].charAt(0).toUpperCase() + temp[1].slice(1);
+                        return temp[0] + ' ' + temp[1];
+                    }
+                };
+
+                var autoCompleter = {
+                    getCompletions: function(editor, session, pos, prefix, callback) {
+                        var curLine = session.getDocument().getLine(pos.row);
+                        var curTokens = curLine.slice(0, pos.column).split(/\s+/);
+                        var valTokens = curLine.slice(0, pos.column).split(':');
+                        if (valTokens[0]) {
+                            var nodeval = valTokens[0].lastIndexOf('(');
+                            if (nodeval > -1) {
+                                var valCmd = valTokens[0].slice(nodeval + 1);
+                            }
+                        }
+                        var curCmd = curTokens[0];
+                        if(!curCmd) return;
+                        var node_pos = nodenames.indexOf(curTokens[curTokens.length-2]);
+                        if (curTokens[curTokens.length-2]) {
+                            var brackets = curTokens[curTokens.length-2].split('(');
+                            if (brackets[1]) {
+                                var colon = brackets[1].split(':')[0];
+                                var colon_pos = nodenames.indexOf(colon);
+                            }
+                            var brack_pos = nodenames.indexOf(brackets[brackets.length-1].slice(0, -1));
+                        }
+                        if (valCmd) {
+                            var val_pos = membernodes.indexOf(valCmd);
```

Comment:
out of scope for val_pos... try declaring everything on the top of the function even if it is unidentified